src/compiler/core.ts

```diff
@@ -429,7 +433,7 @@ namespace ts {
                         addRange(result, v);
                     }
                     else {
-                        result.push(v);
+                        result.push(v as T);

```

Comment:
Is this cast here because the `isArray` helper doesn't guard for readonly array types? Can it not be overloaded to do so?